design/deploy.md

```diff
@@ -0,0 +1,98 @@
+# Deploy
+
+A proposal to add support for hosting service deployment to the Polymer CLI.
+
+## Goals
+- Provide new users with a fast way to get a project hosted online.
+- Provide advanced users with an initial bootstrap for more advanced deployments.
+- Demonstrate the [PRPL](https://www.polymer-project.org/2.0/toolbox/server) pattern where HTTP/2 Server Push is available.
+
+## Non-goals
+- Do not try to support every possible deployment environment and configuration, or even a lot of them. Just enough to get users started.
+
+## Design
+
+- Add an `--environment` flag.
```

Comment:
Please note this issue https://github.com/Polymer/polymer-build/issues/146 where user talks about `environment` in the typical dev/stage/prod forms.  Worth keeping-in-mind so we don't overload the term in a way that might cause any confusions as to users' conventional use.